datadog/dogstatsd/base.py

```diff
@@ -265,6 +282,9 @@ def _send_to_server(self, packet):
         try:
             # If set, use socket directly
             (self.socket or self.get_socket()).send(packet.encode(self.encoding))
+        except socket.timeout:
+            # dogstatsd is overflowing, drop the packets (mimicks the UDP behaviour)
+            return
```

Comment:
Why do we need this? From what I understand, we should never get a timeout on non-blocking mode.